packages/contract/src/test-contract/src/main.sw

```diff
@@ -4,25 +4,25 @@ use std::storage::store;
 use std::storage::get;
 
 abi TestContract {
-  fn initialize_counter(gas_: u64, amount_: u64, coin_: b256, value: u64) -> u64;
-  fn increment_counter(gas_: u64, amount_: u64, coin_: b256, amount: u64) -> u64;
-  fn counter(gas_: u64, amount_: u64, asset_id_: b256, value: ()) -> u64;
+  fn initialize_counter(value: u64) -> u64;
+  fn increment_counter(amount: u64) -> u64;
+  fn counter(value: ()) -> u64;
 }
 
 const COUNTER_KEY = 0x0000000000000000000000000000000000000000000000000000000000000000;
 
 impl TestContract for Contract {
-  fn initialize_counter(gas_: u64, amount_: u64, asset_id_: b256, value: u64) -> u64 {
+  fn initialize_counter(value: u64) -> u64 {
     store(COUNTER_KEY, value);
     value
   }
-  fn increment_counter(gas_: u64, amount_: u64, asset_id_: b256, amount: u64) -> u64 {
+  fn increment_counter(amount: u64) -> u64 {
     let value: u64 = get(COUNTER_KEY);
     let value = value + amount;
     store(COUNTER_KEY, value);
     value
   }
-  fn counter(gas_: u64, amount_: u64, asset_id_: b256, value: ()) -> u64 {
+  fn counter(value: ()) -> u64 {
```

Comment:
`value: ()` can be dropped.

Reply:
I kept this one and also add a new method `counternoparams` as this is the contract package. On the `providers` I have removed the empty param.